controllers/complianceeventsapi/server.go

```diff
@@ -471,6 +505,93 @@ func parseQueryArgs(queryArgs url.Values, isCSV bool) (*queryOptions, error) {
 		}
 	}
 
+	parsed, err := setAuthorizedClusters(ctx, db, parsed, userConfig)
+	if err != nil {
+		// ErrNoAccess needs queryOptions
+		return parsed, err
+	}
+
+	return parsed, nil
+}
+
+// getAuthorizedClusterNames verifies that if a cluster filter is provided,
+// the user has access to this filter. If no cluster filter is provided,
+// it sets the cluster filter to all managed clusters the user has access to.
+func setAuthorizedClusters(ctx context.Context, db *sql.DB, parsed *queryOptions,
+	userConfig *rest.Config,
+) (*queryOptions, error) {
+	unAuthorizedClusters := []string{}
+
+	// Get all managedCluster rules
+	allRules, err := getManagedClusterRules(userConfig, nil)
+	if err != nil {
+		return parsed, err
+	}
+
+	tempClusterIDs, okClusterIds := parsed.Filters["clusters.cluster_id"]
+	clusterIDs := make([]string, len(tempClusterIDs))
+
+	// Deep copy parsed.Filters["clusters.cluster_id"]
+	copy(clusterIDs, tempClusterIDs)
+
+	if okClusterIds {
+		// Convert id to name
+		for index, id := range clusterIDs {
+			clusterName, err := getClusterNameFromID(ctx, db, id)
+			if err != nil {
+				if errors.Is(err, sql.ErrNoRows) {
+					// This leads to send forbidden error to the user
+					parsed.Filters["clusters.cluster_id"] = append(parsed.Filters["clusters.cluster_id"][:index],
+						parsed.Filters["clusters.cluster_id"][index+1:]...)
+
+					continue
+				}
+
+				log.Error(err, "Failed to get cluster name from cluster ID", "ID", id)
+
+				return parsed, err
+			}
+
+			if !getAccessByClusterName(allRules, clusterName) {
+				unAuthorizedClusters = append(unAuthorizedClusters, id)
+			}
+		}
+	}
+
+	parsedClusterNames, okClusterNames := parsed.Filters["clusters.name"]
+	if okClusterNames {
```

Comment:
This if statement is not needed since ranging over an empty/uninitialized slice will just be skipped.